packages/bsky/src/auto-moderator/index.ts

```diff
@@ -0,0 +1,184 @@
+import { ModerationService } from '../services/moderation'
+import { TakedownFlagger } from './abyss'
+import { HiveLabeler, ImgLabeler } from './hive'
+import { KeywordLabeler, TextLabeler } from './keyword'
+import { ids } from '../lexicon/lexicons'
+import { ImageUriBuilder } from '../image/uri'
+import { ImageInvalidator } from '../image/invalidator'
+import { Abyss } from './abyss'
+
+export class AutoModerator {
+  public pushAgent?: AtpAgent
+
+  public takedownFlagger?: TakedownFlagger
+  public imgLabeler?: ImgLabeler
+  public textLabeler?: TextLabeler
+
+  services: {
+    label: (db: PrimaryDatabase) => LabelService
+    moderation?: (db: PrimaryDatabase) => ModerationService
+  }
+
+  constructor(
+    public ctx: {
+      db: PrimaryDatabase
+      idResolver: IdResolver
+      cfg: IndexerConfig
+      backgroundQueue: BackgroundQueue
+      imgUriBuilder?: ImageUriBuilder
+      imgInvalidator?: ImageInvalidator
+    },
+  ) {
+    const { imgUriBuilder, imgInvalidator } = ctx
+    const { hiveApiKey, abyssEndpoint, abyssPassword } = ctx.cfg
+    this.services = {
+      label: LabelService.creator(null),
+    }
+    if (imgUriBuilder && imgInvalidator) {
+      this.services.moderation = ModerationService.creator(
+        imgUriBuilder,
+        imgInvalidator,
+      )
+    } else {
+      log.error(
+        { imgUriBuilder, imgInvalidator },
+        'moderation service not properly configured',
+      )
+    }
+
+    this.imgLabeler = hiveApiKey ? new HiveLabeler(hiveApiKey, ctx) : undefined
+    this.textLabeler = new KeywordLabeler(ctx.cfg.labelerKeywords)
+    if (abyssEndpoint && abyssPassword) {
+      this.takedownFlagger = new Abyss(abyssEndpoint, abyssPassword, ctx)
+    } else {
+      log.error(
+        { abyssEndpoint, abyssPassword },
+        'abyss not properly configured',
+      )
+    }
+
+    if (ctx.cfg.labelerPushUrl) {
+      const url = new URL(ctx.cfg.labelerPushUrl)
+      this.pushAgent = new AtpAgent({ service: url.origin })
+      this.pushAgent.api.setHeader(
+        'authorization',
+        buildBasicAuth(url.username, url.password),
+      )
+    }
+  }
+
+  processRecord(uri: AtUri, cid: CID, obj: unknown) {
+    this.ctx.backgroundQueue.add(async () => {
+      const { text, imgs } = getFieldsFromRecord(obj)
+      await Promise.all([
+        this.labelRecord(uri, cid, text, imgs).catch((err) => {
+          log.error(
+            { err, uri: uri.toString(), record: obj },
+            'failed to label record',
+          )
+        }),
+        this.checkImgForTakedown(uri, cid, imgs).catch((err) => {
+          log.error(
+            { err, uri: uri.toString(), record: obj },
+            'failed to check img for takedown',
+          )
+        }),
+      ])
+    })
+  }
+
+  async labelRecord(uri: AtUri, recordCid: CID, text: string[], imgs: CID[]) {
+    if (uri.collection === ids.AppBskyActorProfile) {
+      // @TODO label profiles
```

Comment:
This all good to leave as a `@TODO`?

Reply:
yup 👍 will work by just deleting that check. just need client signoff that there won't be adverse side effects